web/pdf_find_controller.js

```diff
@@ -193,12 +238,24 @@ class PDFFindController {
       if (matchIdx === -1) {
         break;
       }
-      matches.push(matchIdx);
+
+      let matchIdx2 = this._fixMatchIdx(pageContentDiffs, matchIdx);
+      let matchEnd = matchIdx + queryLen - 1;
+      let matchEnd2 = this._fixMatchIdx(pageContentDiffs, matchEnd);
+
+      matches.push(matchIdx2);
+      matchesLengths.push(matchEnd2 - matchIdx2 + 1);
+    }
+
+    // Prepare arrays for storing the matches.
+    if (!this.pageMatchesLength) {
+      this.pageMatchesLength = [];
```

Comment:
`pageMatches` is initialized in the `reset` method (refer to https://github.com/rossj/pdf.js/blob/1b0e63c19e0c496c51cadb4aaa2d6e5cf0c7d8d6/web/pdf_find_controller.js#L67), so let's do that for this one too to avoid this check here and simplify this a bit. Note though that there already appears to be a member variable named `this.pageMatchesLength`, so I think you need to choose a different name here.

Reply:
You are right that `this.pageMatchesLength` is already in use for the word (non-phrase) search, and also for the actual highlighting in `text_layer_builder.js` [here](https://github.com/mozilla/pdf.js/blob/99060e2485cacb8513e7bc50d88b70101069b1c9/web/text_layer_builder.js#L315). Previously, phrase search did not require tracking separate match lengths, but it does with this update, and by re-using the same member variable in the updated phrase-search function, it prevents us from having to further update `text_layer_builder.js`. Additionally, the current searching method is either phrase matching, or word matching, but not both, so I don't think there is any conflict with the use of `this.pageMatchesLength`.

I agree that the initialization could be handled better. I just copied the initialization from `_calculateWordMatch`, but I think it would be better to update both methods and initialize it as you suggest